mitosheet/src/components/taskpanes/UpdateImports/UpdateImportsTaskpane.tsx

```diff
@@ -0,0 +1,236 @@
+                    return (
+                        <ImportCard 
+                            key={stepImportData.step_id + index}
+                            dataframeCreationIndex={{
+                                step_id: stepImportData.step_id,
+                                index: index
+                            }}
+                            dataframeCreationData={{
+                                step_type: 'excel_import',
+                                file_name: _import.params.file_name,
+                                sheet_name: sheetName
+                            }}
+                            displayedImportCardDropdown={displayedImportCardDropdown}
+                            setDisplayedImportCardDropdown={setDisplayedImportCardDropdown}
+                            setReplacingDataframeState={setReplacingDataframeState}
+                        />
+                    )
+                })
+            } else {
+                return _import.params.df_names.map((dfName, index) => {
+                    return (
+                        <ImportCard 
+                            key={stepImportData.step_id + index}
+                            dataframeCreationIndex={{
+                                step_id: stepImportData.step_id,
+                                index: index
+                            }}
+                            dataframeCreationData={{
+                                step_type: 'dataframe_import',
+                                df_name: dfName,
+                            }}
+                            displayedImportCardDropdown={displayedImportCardDropdown}
+                            setDisplayedImportCardDropdown={setDisplayedImportCardDropdown}
+                            setReplacingDataframeState={setReplacingDataframeState}
+                        />
+                    )
+                })
+            }
+        })
+    })
+
+
+    if (replacingDataframeState === undefined) {
+        return (
+            <DefaultTaskpane>
+                <DefaultTaskpaneHeader 
+                    header="Update Imports"
+                    setUIState={props.setUIState}           
+                />
+                <DefaultTaskpaneBody>
+                    {updateImportCards}
+                </DefaultTaskpaneBody>
+                <DefaultTaskpaneFooter>
+                    <TextButton 
+                        variant="dark"
+                        onClick={() => props.mitoAPI.updateExistingImports(updatedStepImportData)}
+                        disabled={undefined } // TODO, disable this if there is an error
+                    >
+                        <p>
+                            Update Imports
+                        </p>
+                    </TextButton>
+                </DefaultTaskpaneFooter>
+            </DefaultTaskpane>
+        )
+    } else if (replacingDataframeState.screen === 'file_browser') {
```

Comment:
Instead of having one case for file_browser, csv_import, and xlsx_import, I could see having just one case for `file` which handles all of this. It would be structured just like the `FileImportTaskpane` but for updating a file. 

So the structure would be:
- UpdateImportsTaskpane
   - ExistingImports
   - UpdateWithFileTaskpane
      - UpdateFileBrowserTaskpane
      - UpdateCSVImportTaskpane
      - UpdateXLSXImportTaskpane
   - UpdateWithDataframeTaskpane

Reply:
This would allow us to keep track of less state in the top level that I think will make this UpdateImportsTaskpane simpler moving forward. There's a lot of state here that is a bit confusing.